featuretools/tests/synthesis/test_deep_feature_synthesis.py

```diff
@@ -746,6 +747,37 @@ def test_makes_direct_features_through_multiple_relationships(games_es):
                 assert feature_with_name(features, f)
 
 
+def test_stacks_multioutput_features(es):
+    fm, feat = ft.dfs(entityset=es,
+                      target_entity="customers",
+                      agg_primitives=[NumUnique, NMostCommon(n=3)],
+                      trans_primitives=[],
+                      max_depth=3
+                      )
+
+    for i in range(3):
+        f = 'NUM_UNIQUE(sessions.N_MOST_COMMON(log.countrycode)[%d])' % i
+        assert feature_with_name(feat, f)
+
+
+def test_seed_multi_output_feature_stacking(es):
+    threecommon = NMostCommon(3)
+    tc = ft.Feature(es['log']['product_id'], parent_entity=es["sessions"], primitive=threecommon)
+
+    stacked = ft.Feature(tc, parent_entity=es['customers'], primitive=NumUnique)
```

Comment:
This feature is stacking on the multi-output feature without slicing it into MultiOutputFeatures, which I don't think we should allow

The "new_seeds" logic in DFS wouldn't be necessary since the users would have constructed the seed features using the MultiOutputFeature class

We should add checks to AggregationFeature and TransformFeature to catch stacking on a multi-output feature like this.

Reply:
I think this is ok-It does that slicing under the hood, and the output still is the same (i.e. the columns are named correctly). I think it is easier for the user to construct the whole feature, which makes sure that that whole feature gets calculated, and then the program stacks it and creates the MultiOutputFeature class for them, since it is not fully a class.

My main point is that I think that allowing this is beneficial, and having users build multi output features rather than having a multi output feature that, when stacked on, creates the MultiOutputFeatures, would be complicated and harder for them. Also, it might cause some confusion, because the MultiOutputFeature is only created when there is stacking going on.